File: 01_Language/01_Functions/python/array_intersect_key.py
# coding: utf-8

def array_intersect_key(d, *args):
    keys = d.keys()
    for arg in args:
        keys &= arg.keys()
    return dict((k, d[k]) for k in keys)


if __name__ == '__main__':
    array1 = {'blue': 1, 'red': 2, 'green': 3, 'purple': 4}
    array2 = {'green': 5, 'blue': 6, 'yellow': 7, 'cyan': 8}
    print(array_intersect_key(array1, array2))

File: 01_Language/01_Functions/js/array_intersect_key.js
function array_intersect_key(arr) {
    let res = {}
    for (let i in arr) {
        let exists = false
        for (let j = 1; j < arguments.length; j++) {
            exists = false
            for (let k in arguments[j]) {
                if (i === k) {
                    exists = true
                    break
                }
            }
            if (!exists) {
                break
            }
        }
        if (exists) {
            res[i] = arr[i]
        }
    }
    return res
}

let array1 = {'blue': 1, 'red': 2, 'green': 3, 'purple': 4}
let array2 = {'green': 5, 'blue': 6, 'yellow': 7, 'cyan': 8}
console.log(array_intersect_key(array1, array2));
